### ISSUE TITLE
Incorrect Use of is_single() and is_page()

### ISSUE THREAD

**Issue by rhurling** [Automattic/wp-super-cache#76]
Hi,
my Multisite debug.log kept getting filled with PHP Notices and I finally decided to further investigate.
It seems that WP Super Cache is using is_single() and is_page() incorrectly in `wp-cache-phase2.php`.

This is what the [Log Deprecated Notices](https://wordpress.org/plugins/log-deprecated-notices/) Plugin says:

```
Incorrect Use of is_single()
Used in wp-content/plugins/wp-super-cache/wp-cache-phase2.php on line 1277.
```

and

```
Incorrect Use of is_page()
Used in wp-content/plugins/wp-super-cache/wp-cache-phase2.php on line 1277.
```

See: https://github.com/Automattic/wp-super-cache/blob/master/wp-cache-phase2.php#L1277

The PHP Notice seems to be appearing on every page load which hits WP Super Cache.

**Comment by kraftbj** [comment]
Thanks for the report. I'll dig into it next set of cycles I can.

**Comment by jom** [comment]
This was fixed in #367.

ref: 
Function: `wp_cache_post_id()`
This line at the time of report:
https://github.com/Automattic/wp-super-cache/blob/0ded455aafce7f49905cad6547de4fd0e8d8cb92/wp-cache-phase2.php#L1277

The current version of that line:
https://github.com/Automattic/wp-super-cache/blob/30714d44cba34f9c537bede2a34922c4ec1e4b99/wp-cache-phase2.php#L1696
